eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/NestedBundle.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017 SchedJoules
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.Keys;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.NullSafe;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class NestedBundle extends AbstractCachingOptional<Bundle>
+{
+    public NestedBundle(final Intent intent)
+    {
+        super(new Factory<Optional<Bundle>>()
+        {
+            @Override
+            public Optional<Bundle> create()
+            {
+                return new NullSafe<>(intent.getBundleExtra(Keys.NESTED_BUNDLE));
```

Comment:
Hmm. good point.
We could do that at least for `Boxable`:
```
<T extends Boxable> BuilderAdapter withExtra(Key<T> key, Optional<T> value);
```

And for Optional Boxes
```
<T> BuilderAdapter withExtra(Key<T> key, Optional<Box<T>> value);
```

That doesn't help here though. But I'm wondering if there isn't another way. Like a `BoxedExtra` that returns the boxed Object instead of the unboxed one. `BoxedExtra<T>` would be an `Optional<Box<T>>` and could be used with the method above. Or am I missing something?

I'm not a big fan of the `withCopiedParam` method. The use case looks way to limited to me. It doesn't work that well when I want to copy a Fragment Extra into an Intent or vice versa.